### PR TITLE
Fix aircraft repulsing other aircraft when not being repulsable themselves

### PR DESCRIPTION
**Pull request by abcdefg30**

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs [hunk 1]
```diff
@@ -200,6 +200,7 @@ public virtual WVec GetRepulsionForce()
 
 			var repulsionForce = self.World.FindActorsInCircle(self.CenterPosition, Info.IdealSeparation)
 				.Where(a => !a.IsDead && a.Info.HasTraitInfo<AircraftInfo>()
+					&& a.Info.TraitInfo<AircraftInfo>().Repulsable
 					&& a.Info.TraitInfo<AircraftInfo>().CruiseAltitude == Info.CruiseAltitude)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: This now does three queries for `AircraftInfo`, which isn't very efficient.  Can you change this to something like the following?

```csharp

.Where(a =>
{
    if (a.IsDead)
        return false;
    var ai = a.Info.TraitInfoOrDefault<AircraftInfo>();
    return ai != null && ai.Repulsable && ai.CruiseAltitude == Info.CruiseAltitude;
})
```

**Comment by abcdefg30** [comment]
Updated.

**Comment by ScottNZ** [comment]
The current behaviour was intentional. What does this solve?

**Comment by penev92** [comment]
I guess we can turn off their repulsion when trying to land? Currently having more than one helicopter and just 1 helipad means they won't rearm 99% of the time, unless micro-ed individually.

**Comment by pchote** [comment]
@ScottNZ maybe intended as a more general solution to #9618?

**Comment by abcdefg30** [comment]
> The current behaviour was intentional. What does this solve?

This is indeed "a more general solution to #9618".
It seems a bit odd to me too, if an actor that is not repulsable itself repulses other actors...

**Comment by penev92** [comment]
:+1:

**Comment by Mailaender** [comment]
Can't spot regressions :white_check_mark: and appreciate the trait lookup save.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/08771babf362d235225cc3849b1f1f8d40bd886f...8258f7f661e433103a39a60a650cc008928171d5)

**Pull request merged**